helix-term/src/commands.rs

```diff
@@ -3419,27 +3420,50 @@ fn open(cx: &mut Context, open: Open) {
             (0, 0)
         } else {
             (
-                line_end_char_index(&text, line_num),
+                line_end_char_index(&doc_text, line_num),
                 doc.line_ending.len_chars(),
             )
         };
 
-        let indent = indent::indent_for_newline(
-            doc.language_config(),
-            doc.syntax(),
-            &doc.config.load().indent_heuristic,
-            &doc.indent_style,
-            doc.tab_width(),
-            text,
-            line_num,
-            line_end_index,
-            cursor_line,
-        );
+        let comment_token = {
+            let tokens = doc
+                .language_config()
+                .and_then(|config| config.comment_tokens.as_ref());
+
+            get_comment_token(doc.text(), tokens, cursor_line)
+        };
+        let new_line_will_be_comment = config.continue_comments && comment_token.is_some();
+
+        let indent = {
+            let line = doc_text.line(cursor_line);
```

Comment:
style nit: we try to avoid using blocks for bindings like this. It's fine for `line` to escape into the outer scope